lib/src/gantt_view.dart

```diff
@@ -226,9 +226,9 @@ class GanttChartViewState extends State<GanttChartView> {
                             ),
                           ),
                           child: Center(
-                            child: Text(
-                              event.getDisplayName(context),
```

Comment:
it would be better to change event.getDisplayName to return a widget instead

Reply:
I actually evaluated both approaches but I think all view-related stuff should be confined in `gantt_view.dart`. Having event.getDisplayName return a Widget means you should add a Text() widget to the event itself, which is not good to me.

Reply:
but we are already using a widget as return for displayNameBuilder in the event

Reply:
Yes, but that's only a "type", it's not actually deciding what to print. If you write something like

```dart 
Widget getDisplayName(context) => 
    displayNameBuilder?.call(context) ??
    Text(event.displayName ?? "")
```

then you have part of the UI in gantt_view.dart and part in event.dart which looks not good to me.
Let's say tomorrow you want to change the text size based on zoom, or any other UI-related stuff, you'll have to change both gantt_view.dart and event.dart.

BTW, this is just IMHO.

Reply:
but don't we already have `stickyAreaEventBuilder` ? it can be overridden to change how the entire widget is displayed (including borders and the background).
we can add a new variable to `GanttChartView` e.g. 

```dart
final Widget Function(BuildContext context, int eventIndex, GanttEventBase event, Color eventColor)? stickyAreaEventNameBuider;
```
which will handle only the display name part

Reply:
The main reason `String Function(BuildContext context)? displayNameBuilder` exists is for localization, which needs access to the `BuildContext`, but UI customization should stay inside gantt view